firefox_puppeteer/ui/pageinfo/deck.py

```diff
@@ -0,0 +1,208 @@
+    def feed(self):
+        """The :class:`FeedPanel` instance for the feed panel.
+
+        :returns: :class:`FeedPanel` instance.
+        """
+        return self._create_panel_for_id('feedPanel')
+
+    @property
+    def general(self):
+        """The :class:`GeneralPanel` instance for the general panel.
+
+        :returns: :class:`GeneralPanel` instance.
+        """
+        return self._create_panel_for_id('generalPanel')
+
+    @property
+    def media(self):
+        """The :class:`MediaPanel` instance for the media panel.
+
+        :returns: :class:`MediaPanel` instance.
+        """
+        return self._create_panel_for_id('mediaPanel')
+
+    @property
+    def panels(self):
+        """List of all the :class:`Panel` instances of the current deck.
+
+        :returns: List of :class:`Panel` instances.
+        """
+        panels = self.element.find_elements(By.TAG_NAME, 'vbox')
+        return [self._create_panel_for_id(panel.get_attribute('id')) for panel in panels
+                if panel.get_attribute('id').endswith('Panel')]
+
+    @property
+    def permissions(self):
+        """The :class:`PermissionsPanel` instance for the permissions panel.
+
+        :returns: :class:`PermissionsPanel` instance.
+        """
+        return self._create_panel_for_id('permPanel')
+
+    @property
+    def security(self):
+        """The :class:`SecurityPanel` instance for the security panel.
+
+        :returns: :class:`SecurityPanel` instance.
+        """
+        return self._create_panel_for_id('securityPanel')
+
+    # Properties for helpers when working with the deck #
+
+    @property
+    def selected_index(self):
+        """The index of the currently selected panel.
+
+        :return: Index of the selected panel.
+        """
+        return int(self.element.get_attribute('selectedIndex'))
+
+    @property
+    def selected_panel(self):
+        """A :class:`Panel` instance of the currently selected panel.
+
+        :returns: :class:`Panel` instance.
+        """
+        return self.panels[self.selected_index]
+
+    # Methods for helpers when working with the deck #
+
+    def select(self, panel):
+        """Selects the specified panel via the tab element.
+
+        :param panel: The panel to select.
+
+        :returns: :class:`Panel` instance of the selected panel.
+        """
+        panel.tab.click()
+        Wait(self.marionette).until(lambda _: panel.selected)
+
+        return panel
+
+
+class Panel(UIBaseLib):
+
+    def __eq__(self, other):
+        return self.element.get_attribute('id') == other.element.get_attribute('id')
+
+    @property
+    def selected(self):
+        """Checks if this panel is selected.
+
+        :returns: True, if the panel is selected.
+        """
+        return self.parent.selected_panel == self
+
+    def select(self):
+        """Selects this panel.
+
+        :returns: :class:`Panel` instance to the this panel.
+        """
+        return self.parent.select(self)
```

Comment:
This will never work for e.g. window classes. Keep in mind that each instance of whatever type has to know about the window it lives in, and e.g. how to find properties, dtd entities. Also I do not understand what you mean with "accessing an element by its parent"? If you are talking about the DOM, its parent will not match our parent, and even doesn't necessarily mean it is the parent we want. So please explain what you have in mind.